src/main/java/io/r2dbc/postgresql/PostgresqlServerErrorException.java

```diff
@@ -347,11 +348,27 @@ public String toString() {
     }
 
     static void handleErrorResponse(BackendMessage message, SynchronousSink<BackendMessage> sink) {
```

Comment:
Probably we want to move that code into an `ExceptionFactory`, wdyt @nebhale? 

The issue here is that we're creating a cycle between `PostgresqlServerErrorException ` and `PostgresqlAuthenticationFailure`. I think we can take care of this during the merge.

Reply:
I could create a Factory, it is not a problem.

I thought about adding one, but I tried to touch the minimum as possible. I did not want to update too much code.

Reply:
In general, that's a good guideline to touch as little as necessary. In this case, it would make sense to go the additional step in changing how exceptions are created.

Reply:
Hello, I just pushed the update with the factory.